src/Elasticsearch/Metadata/Resource/Factory/ElasticsearchProviderResourceMetadataCollectionFactory.php

```diff
@@ -42,6 +46,15 @@ public function create(string $resourceClass): ResourceMetadataCollection
 
             if ($operations) {
                 foreach ($resourceMetadata->getOperations() as $operationName => $operation) {
+                    if (null !== ($elasticsearch = $operation->getElasticsearch())) {
+                        $solution = $elasticsearch
+                            ? sprintf('Pass an instance of %s to $persistenceMeans instead', ElasticsearchDocument::class)
+                            : 'You will have to remove it when upgrading to v4';
+                        trigger_deprecation('api-platform/core', '3.1', sprintf('Setting "elasticsearch" in Operation is deprecated. %s', $solution));
+                    }
+                    if (false === $elasticsearch && $operation->getPersistenceMeans() instanceof ElasticsearchDocument) {
+                        throw new \LogicException('You cannot set $elasticsearch to false while configuring an elasticsearch document');
```

Comment:
IMO the previous deprecation is fine, we can remove this exception and make it work anyways.